lib/logproto/logproto-server.c

```diff
@@ -122,12 +122,20 @@ log_proto_server_init(LogProtoServer *self, LogTransport *transport, const LogPr
   self->transport = transport;
 }
 
-void
+gboolean
 log_proto_server_options_set_encoding(LogProtoServerOptions *self, const gchar *encoding)
 {
-  if (self->encoding)
-    g_free(self->encoding);
+  GIConv convert;
+
+  g_free(self->encoding);
   self->encoding = g_strdup(encoding);
+
+  /* validate encoding */
+  convert = g_iconv_open("utf8", encoding);
```

Comment:
I agree that code should be under scrutiny and going back and finding how a
bug could have been prevented is a worthwhile exercise.

However it'd have been nice if you provided this context in the note. I
didn't know that there was a failure of some kind. I still don't get what
the bug is that triggered this comment as without  that I cant judge the
recommended solution.

You surprise me that iconv wouldnt understand "utf8" as an encoding. I use
that pretty regularly and so far it worked just fine, but maybe I was
completely mislead.

I still think that adding a define and using it everywhere will not resolve
the issue, because:

1) then you'll have to remember that there's this constant. and newcomers
won't.

2) in reviews you'd want to flag errors if that  happens, thus everyone
will have to focus on that

3) if you focus on these kind of things you'll not get to the real
important stuff, like layering, interfaces and duplicate implementations

The solution should be a failing in test. And again, can you pls tell me
what failed which triggered this review so we can fix the test instead?

Thanks
On May 9, 2016 10:47 PM, "Tamas Nagy" <notifications@github.com> wrote:

> In lib/logproto/logproto-server.c
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62569998>:
>
> >    self->encoding = g_strdup(encoding);
> > +
> > +  /* validate encoding */
> > +  convert = g_iconv_open("utf8", encoding);
>
> In my opinion, we should review merged code just as often as new code, but
> that's besides the point. The correct name of the encoding is "UTF-8" as
> defined in the standards: https://www.ietf.org/rfc/rfc2279.txt
> http://standards.iso.org/ittf/PubliclyAvailableStandards/index.html
> (ISO/IEC 10646) https://en.wikipedia.org/wiki/UTF-8#External_links
> libiconv, for example, won't accept "utf8" as an encoding at all. If we had
> extracted this constant earlier, this typo would not have occurred.
> http://git.savannah.gnu.org/cgit/libiconv.git/tree/lib/encodings.def?id=48f31c744eb14da699c6ba3fd03f5f232c476bbb#n60
> … <#m_-5703237840663958278_>
> On Mon, May 9, 2016 at 5:53 PM, Balazs Scheidler ***@***.***> wrote: In
> lib/logproto/logproto-server.c <#797 (comment)
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62523578>>: >
> self->encoding = g_strdup(encoding); > + > + /* validate encoding */ > +
> convert = g_iconv_open("utf8", encoding); Hmm, this is already merged. But
> I don't agree that it matters how we call utf8. There are aliases for that
> for a reason. It doesn't make stuff more readable, reliable or anything. On
> the other hand it takes away energy to look after/validate these kind of
> mistakes we would make while coding. … <#m_-3246939082143047282_> On May 9,
> 2016 5:36 PM, "Tamas Nagy" ***@***.***> wrote: In
> lib/logproto/logproto-server.c <#797
> <https://github.com/balabit/syslog-ng/pull/797> (comment) <#797 (comment)
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62520468>>>: >
> self->encoding = g_strdup(encoding); > + > + /* validate encoding */ > +
> convert = g_iconv_open("utf8", encoding); It would look nicer if we used
> UTF-8 everywhere, and we could introduce a constant for that as well. — You
> are receiving this because you were mentioned. Reply to this email directly
> or view it on GitHub <
> https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62520468>
> — You are receiving this because you commented. Reply to this email
> directly or view it on GitHub <
> https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62523578>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62569998>
>